aten/src/ATen/cpu/vec256/vec256_base.h

```diff
@@ -316,6 +316,9 @@ struct Vec256 {
     }
     return ret;
   }
+  Vec256<T> polar() const {
+    return *this;
+  }
```

Comment:
Yes, in a template class you need to re-implement all methods during specialization`.

I think the default case should return `abs(x)`, not `real(x)`.

If you are publicly adding a new kernel, you also need to specify the derivate so that Autograd works in [derivatives.yaml](https://github.com/pytorch/pytorch/blob/f1efe5102876958976b648691ffde422d85b6a0e/tools/autograd/derivatives.yaml). You would have to combine the derivative of `abs()` and `angle()` in some way.

Reply:
Oh, yes you're right. I didn't know about `derivates.yaml`, will look into it.